src/test/java/org.spongepowered.mod/test/integration/regression/CrossDimensionCommandBlock.java

```diff
@@ -0,0 +1,110 @@
+import org.spongepowered.api.world.difficulty.Difficulties;
+import org.spongepowered.api.world.storage.WorldProperties;
+import org.spongepowered.mctester.internal.BaseTest;
+import org.spongepowered.mctester.api.junit.MinecraftRunner;
+import org.spongepowered.mctester.junit.TestUtils;
+import org.spongepowered.mod.test.integration.RegressionTest;
+
+import java.util.Optional;
+
+@RunWith(MinecraftRunner.class)
+@RegressionTest(ghIssue = "https://github.com/SpongePowered/SpongeCommon/issues/1718")
+public class CrossDimensionCommandBlock extends BaseTest  {
+
+    public CrossDimensionCommandBlock(TestUtils testUtils) {
+        super(testUtils);
+    }
+
+    @Test
+    public void testUseCommandBlock() throws Throwable {
+
+        Location<World> targetLocation = this.testUtils.runOnMainThread(() -> {
+
+            //Sponge.getCommandManager().process(Sponge.getServer().getConsole(), "op " + McTester.getThePlayer().getName());
+
+            WorldProperties worldProperties = Sponge.getServer().createWorldProperties("the_command_block_world",
+                    WorldArchetype.builder().commandsAllowed(true).difficulty(Difficulties.PEACEFUL)
+                            .dimension(DimensionTypes.OVERWORLD)
+                            .enabled(true)
+                            .gameMode(GameModes.CREATIVE)
+                            .generator(GeneratorTypes.FLAT)
+                            .build("mctester:command_block_world", "Command block World"));
+
+            World world = Sponge.getServer().loadWorld(worldProperties).get();
+
+            this.testUtils.getThePlayer().setLocation(world.getSpawnLocation());
+            this.testUtils.getThePlayer().setItemInHand(HandTypes.MAIN_HAND, ItemStack.of(ItemTypes.COMMAND_BLOCK, 1));
+
+            Location<World> target = this.testUtils.getThePlayer().getLocation().add(0, 1, 1);
+            return target;
+        });
+
+        this.testUtils.waitForInventoryPropagation();
+
+        this.client.lookAt(targetLocation.getPosition());
+
+        if (!this.testUtils.runOnMainThread(targetLocation::getBlockType).equals(BlockTypes.AIR)) {
+            this.client.leftClick();
+        }
+
+        // Right click once to place the command block, then right click again to open it
+        this.client.rightClick();
+        this.client.rightClick();
+
+        Optional<String> guiClass = this.client.getOpenGuiClass();
+
+        assertThat("Failed to open command block gui!", guiClass, Matchers.equalTo(Optional.of("net.minecraft.client.gui.GuiCommandBlock")));
+
+        /*this.testUtils.runOnMainThread(() -> {
+            InventoryArchetype type = McTester.getThePlayer().getOpenInventory().orElseThrow(() -> new AssertionError("Failed to open command block gui!")).getArchetype();
+
+            assertThat(type, Matchers.equalTo(InventoryArchetypes.)
+
+            Assert.assertThat(McTester.getThePlayer().getOpenInventory().orElse(null), Matcher);
+        });*/
```

Comment:
Yes, command block gui does not have an inventory. Can we just use forge code here to check the gui type open? `Minecraft.getMinecraft.openGui`

Reply:
I've deliberately avoided touching `Minecraft` in any of the test classes. See https://github.com/Aaron1011/McTester/issues/2 for an explanation.